lib/astarte/common/generators/timestamp.ex

```diff
@@ -0,0 +1,42 @@
+#
+# This file is part of Astarte.
+#
+# Copyright 2025 SECO Mind Srl
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+defmodule Astarte.Common.Generators.Timestamp do
+  use ExUnitProperties
+
+  @moduledoc """
+  Unix timestamp generator
+  """
+  @min_default 0
+  @max_default 2_556_143_999
+
+  @doc """
+  Generates a random timestamp between min and max, defaulting to 0 and 2_556_143_999.
+  """
+  @spec timestamp([{:min, integer()} | {:max, integer()}]) :: StreamData.t(integer())
```

Comment:
Nit: this means that `timestamp/1` always needs to be called with either `min` or `max`, but the line below states that opts is empty if not explicitly given. A more complete typespec could be something like
```suggestion
  @type timestamp_opts :: {:min, integer()} | {:max, integer()}
  @spec timestamp([timestamp_opts]) :: StreamData.t(integer())
  @spec timestamp([]) :: StreamData.t(integer())
```

Reply:
While I agree that adding a type for the option is more readable, note you typespec and gabriele's are equivalent: it's a list of _type_ `timestamp_opts`, not with one element of that type.
For this reason you also don't need to make the empty list a special case

Reply:
There is no need to spend more time here, but they are not equivalent: the previous spec describes a `timestamp/1` function that accepts a (possibly empty) keyword list; my suggestion describes the same `timestamp/1` function and a `timestamp/0` one - which is what happens when using default arguments